dotnet/src/Functions/Functions.OpenApi/Extensions/RestApiOperationExtensions.cs

```diff
@@ -38,10 +38,10 @@ public static IReadOnlyList<RestApiParameter> GetParameters(
             parameters.AddRange(GetPayloadParameters(operation, addPayloadParamsFromMetadata, enablePayloadNamespacing));
         }
 
-        // Create a property alternative name without special symbols that are not supported by SK template language.
+        // Set parameter argument name, if not already set, that contain only supported by SK template language symbols.
         foreach (var parameter in parameters)
         {
-            parameter.AlternativeName = InvalidSymbolsRegex().Replace(parameter.Name, "_");
+            parameter.ArgumentName ??= InvalidSymbolsRegex().Replace(parameter.Name, "_");
```

Comment:
The comment is rephrased because it was misleading. In short, this functionality is not related to templates, it was introduced to replace dashed in OpenAPI parameters and atm is kept for backward compatibility only.